### ISSUE TITLE
Building a project fails

### ISSUE THREAD

**Issue by mit-mit** [flutter/flutter-intellij#945]
## Steps to Reproduce

1. Create a new Flutter project inside IntelliJ
1. Invoke Build > Build Project

Fails with a bunch of Android errors:

```
Information:java: Errors occurred while compiling module 'android'
Information:javac 1.8.0_72 was used to compile java sources
Information:24/04/2017, 15.28 - Compilation completed with 1 error and 3 warnings in 2s 431ms
Warning:java: source value 1.3 is obsolete and will be removed in a future release
Warning:java: target value 1.3 is obsolete and will be removed in a future release
Warning:java: To suppress warnings about obsolete options, use -Xlint:-options.
/Users/mit/tmp/test31/android/app/src/main/java/com/yourcompany/test31/MainActivity.java
Error:(10, 6) java: annotations are not supported in -source 1.3
  (use -source 5 or higher to enable annotations)
```

**Comment by devoncarew** [comment]
I think your Java source compatibility level is set to 1.3. It should probably be set to 1.8. @skybrian, is this something we can add to the project metadata?

**Comment by devoncarew** [comment]
Does setting the 'Sources' setting from here: https://flutter.io/intellij-ide/#edit-android-code help?

**Comment by mit-mit** [comment]
Oh, we changed the default Project build action to build the android sub-module not the main Flutter module? It seems like Build > Build Project should correspond to `flutter build`!?

**Comment by skybrian** [comment]
Before, Java wasn't enabled at all. Now it is.

I don't know what Build > Build Project did before, if anything.

**Comment by skybrian** [comment]
I created a new project, and for me the project settings were:
```
   Project SDK: <No SDK>
   Project language level: <8 - Lambdas, type annotations, etc.>
```

The language level wasn't explicitly set in .ideas/misc.xml, so I'd guess it's picking up a default value from somewhere else.

Running Build->Build Project creates .class files in out/production/android, but it doesn't do anything else as far as I can tell.

**Comment by devoncarew** [comment]
The IntelliJ `Build > Build Project` action doesn't really have any meaning or behavior for Dart or Flutter projects.

**Comment by pq** [comment]
> The IntelliJ Build > Build Project action doesn't really have any meaning or behavior for Dart or Flutter projects.

Given this, I wonder if we can't intercept the action and disable/ignore or otherwise tailor a message for it?

cc @alexander-doroshko

**Comment by sethladd** [comment]
> action doesn't really have any meaning or behavior for Dart or Flutter projects.

Would that mean "flutter build" ?

**Comment by alexander-doroshko** [comment]
> The IntelliJ Build > Build Project action doesn't really have any meaning or behavior for Dart or Flutter projects.

> Given this, I wonder if we can't intercept the action and disable/ignore or otherwise tailor a message for it? cc @alexander-doroshko

Indeed, Build > Build Project does nothing for Dart/Flutter, but works for Android's Java.

'Build Project' action is designed to do compilation required to start an app. No compilation required to start Dart or Flutter, so this should remain a noop. 'Dart build' or 'flutter build' should not be invoked on  Build > Build Project, these actions look more like 'publish', not like 'compile'. WebStorm and other small IDEs do not have Build menu at all because in Web world there's no compilation similar to Java.

This bug report is about problems with Android's Java compilation, it's not about Dart or Flutter. So you should look for the solution around Android module configuration.

**Comment by skybrian** [comment]
@alexander-doroshko  I'm considering creating a new module type, "Flutter Android", which is only a slight variation of a Java module type. Do you think this is feasible?

Currently we generate a Java module for the "android" subdirectory, and then users are prompted to convert it to an Android module, but this doesn't seem like the best user experience. (Even if we got it basically working, it wouldn't support Flutter's hot reload.)

**Comment by alexander-doroshko** [comment]
I didn't dig deep into the problem but introducing a new module type seems to be a too heavy solution that may bring more problems than solve.

**Comment by pq** [comment]
If we want to support `flutter build` I suggest we add a new menu item in the **Flutter** top level tool menu.  That still leaves open what build should do here...  Maybe nothing?

**Comment by skybrian** [comment]
I think at this point is does nothing other than check that the Java code compiles. That should work provided that the project Java SDK is set properly.

**Comment by mit-mit** [comment]
Flutter is *our* module type, why can we not make building that module type the same as `flutter build`?

**Comment by alexander-doroshko** [comment]
> Flutter is our module type, why can we not make building that module type the same as flutter build?

As a separate action in a Flutter menu - I have no opinion on the question.
As an extension of standard Main Menu Build > Build Project action - I see no reasons to do it, and there are reasons not to do it.

**Comment by devoncarew** [comment]
Revisiting this, I think the behavior we want is for users to be able to build Android and iOS release artifacts of their app. The `Build > Build Project` menu item may not do what we want here - there's no way to specify an ios or apk build, or build flavors.

We may want to use the Build menu, and have custom Flutter build ios and build apk menu items, or look into using IntelliJ build targets.

In any case, I'll re-purpose this bug to something like 'support building of ios and android release artifacts'.
